src/web/mod.rs

```diff
@@ -358,13 +371,18 @@ fn match_version(conn: &mut Client, name: &str, version: Option<&str>) -> Option
     // semver is acting weird for '*' (any) range if a crate only have pre-release versions
     // return first non-yanked version if requested version is '*'
     if req_version == "*" {
-        return versions_sem.first().map(|v| MatchVersion {
-            corrected_name,
-            version: MatchSemver::Semver((v.0.to_string(), v.1)),
-        });
+        return versions_sem
+            .first()
+            .map(|v| MatchVersion {
+                corrected_name,
+                version: MatchSemver::Semver((v.0.to_string(), v.1)),
+            })
+            .ok_or(Nope::CrateNotFound);
```

Comment:
```suggestion
            .ok_or(Nope::VersionNotFound);
```

Reply:
Please also add a test for this case by requesting `/:crate/*` for a crate that only has pre-releases, then make sure that it fails before and succeeds after this change.

Reply:
Such a request would return the correct crate before and after the PR. The difference in behavior is for a `/:crate/*` request for a crate that only has yanked versions in which case `match_version` would have returned a None and the resulting page would have displayed `no such resource`.  
I've added a test case asserting that in this case the correct `VersionNotFound` error is returned.